kroxylicious/src/main/java/io/kroxylicious/proxy/filter/FilterChainFactory.java

```diff
@@ -5,16 +5,100 @@
  */
 package io.kroxylicious.proxy.filter;
 
+import java.lang.reflect.InvocationTargetException;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
 /**
  * Abstracts the creation of a chain of filter instances, hiding the configuration
  * required for instantiation at the point at which instances are created.
  * New instances are created during initialization of a downstream channel.
  */
-public interface FilterChainFactory {
+public class FilterChainFactory {
+
+    public static class Builder {
+
+        private final List<FilterChainContribution> contributions = new ArrayList<>();
+
+        public Builder with(String contributorClassName) {
+            throw new UnsupportedOperationException();
+        }
+
+        public Builder with(String contributorClassName, Map<String, String> configuration) {
+            throw new UnsupportedOperationException();
+        }
```

Comment:
These two would be used when everything is driven by a text-based external configuration format, whatever that will be.